pandas-stubs/core/indexes/base.pyi

```diff
@@ -393,7 +393,7 @@ class Index(IndexOpsMixin[S1]):
     def asof_locs(self, where, mask): ...
     def sort_values(self, return_indexer: bool = ..., ascending: bool = ...): ...
     def sort(self, *args, **kwargs) -> None: ...
-    def shift(self, periods: int = ..., freq=...) -> None: ...
+    def shift(self, periods: int = ..., freq=...) -> Self: ...
```

Comment:
One small question here, `pd.Index.shift` is technically not implemented so wondering if it should be `NoReturn` here and `Self` for `pd.DatetimeIndex`, `pd.TimedeltaIndex` and `pd.PeriodIndex` or if this is fine as is (technically it would fail at runtime anyway so the typing would not help much to spot the bug).

Reply:
I think we should take `shift()` out of here and only include it in the stubs for those 3 types.

Reply:
Agreed! Thanks for the direction